### PR TITLE
Add olminstall ITS smoke pipeline and contributor guide

### PR DESCRIPTION
**Pull request by manosnoam**
https://redhat.atlassian.net/browse/RHOAIENG-57714

- Add `integration-tests/olminstall/` — a Konflux IntegrationTestScenario that provisions an ephemeral EaaS HyperShift cluster, installs the RHOAI/ODH operator via OLM from a Konflux-built FBC fragment, and runs basic install verification.
- Pipeline handles IDMS mirroring (`registry.redhat.io/rhoai` → `quay.io/rhoai`), Quay pull-secret injection via HCCO global-pull-secret-syncer, AllNamespaces OperatorGroup, and detailed CSV/deployment result reporting.
- `run-olminstall.sh` CLI helper: auto-resolves latest FBCF image, applies ITS with optional overrides (`--konflux-repo`, `--channel`, `--product odh`), creates a Snapshot, waits for the PipelineRun, and streams logs via `tkn`.
- ITS manifests for both tenants: `its-olminstall-open-data-hub-tenant.yaml` (ODH, `optional: true`) and `its-olminstall-rhoai-tenant.yaml` (sandbox/manual trigger).
- Add `doc/contributing-konflux-testing-rhoai.md` — practical guide for testing Konflux `integration-tests/` before pushing: glossary, `oc` / `tkn` workflows, Snapshot-driven test triggers, multi-system (Konflux → GitHub Actions → Jenkins) debugging order, and pre-review checklist.
- Add `integration-tests/olminstall/README.md` with glossary linking, tool requirements, triggering docs, and `run-olminstall.sh` usage examples.

### Pipeline example for running:
`./olminstall/run-olminstall.sh --konflux-repo https://github.com/manosnoam/odh-konflux-central.git --konflux-branch add-olminstall-its --channel beta --product rhoai --version 3.4`

image

image

## Summary by CodeRabbit

* **New Features**
  * Added a Tekton smoke test pipeline, installer and helper scripts, and a CLI to trigger/monitor operator installation tests.

* **Tests**
  * Added PipelineRun and Snapshot manifests plus IntegrationTestScenario entries for automated and manual smoke runs.

* **Documentation**
  * Added a contributor guide and detailed README with run/debugging instructions and troubleshooting workflows.

* **Chores**
  * Updated .gitignore to ignore editor metadata and normalize file endings.

### CODE REVIEW

File: doc/contributing-konflux-testing-rhoai.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+tkn pipelinerun logs <name> -n rhoai-tenant -f
+```
+
+3. **GitHub Actions** — open the `conforma-reporter/smoke-trigger.yaml` workflow run
+   triggered by the dispatch and confirm expected inputs are present.
+   (`conforma-reporter` is an internal repo that bridges Konflux → Jenkins: its
+   `smoke-trigger.yaml` workflow dispatches the nightly Jenkins smoke job.)
+
+4. **Jenkins** — open the triggered `autotrigger-smoke` job and confirm
+   parameters match the values from step 2.
```
[the remark this answers is not in the document]
Comment (manosnoam) [reply doc/contributing-konflux-testing-rhoai.md]: fixed

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+              fi
+      params:
+        - name: FBCF_IMAGE
+          value: $(tasks.extract-fbcf-image.results.FBCF_IMAGE)
+        - name: TEST_STATUS
+          value: $(tasks.install-operator.results.INSTALL_STATUS)
+        - name: OPERATOR_VERSION
+          value: $(tasks.install-operator.results.OPERATOR_VERSION)
+        - name: PIPELINE_RUN_NAME
+          value: $(context.pipelineRun.name)
```
[the remark this answers is not in the document]
Comment (manosnoam) [reply integration-tests/olminstall/olminstall-smoke-pipeline.yaml:600]: fixed

File: integration-tests/olminstall/run-olminstall.sh [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+
+# ── Check for an already-running olminstall PipelineRun ───────────────────────
+echo "Checking for running olminstall PipelineRun..."
+PR=$(oc get pipelineruns -n "${NAMESPACE}" -o json 2>/dev/null \
+  | jq -r '
+      .items[]
+      | select(.metadata.name | test("olminstall"))
+      | select(.status.completionTime == null)
+      | .metadata.name' \
+  | head -1 || true)
```
[the remark this answers is not in the document]
Comment (manosnoam) [reply integration-tests/olminstall/run-olminstall.sh:183]: fixed

File: doc/contributing-konflux-testing-rhoai.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+## Log in and pick a namespace
+
+Use your team-approved Konflux/OpenShift endpoints:
+
+| Context                                          | Konflux UI                                 | OpenShift API                       |
+| ------------------------------------------------ | ------------------------------------------ | ----------------------------------- |
+| Replace `<cluster-domain>` with your team's host | `https://konflux-ui.apps.<cluster-domain>` | `https://api.<cluster-domain>:6443` |
+
+> Cluster URLs may change over time. Verify at [konflux-ci.dev](https://konflux-ci.dev) for the actual domain.
```
Comment (ckhordiasma) [inline doc/contributing-konflux-testing-rhoai.md]: cluster info is actually at https://konflux.pages.redhat.com/docs/users/cluster-info/cluster-info.html

File: doc/contributing-konflux-testing-rhoai.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+You need permission to create the resources your test uses: at minimum a
+[PipelineRun](#pipelinerun), and for Snapshot-driven tests also a [Snapshot](#snapshot) and an
+[IntegrationTestScenario](#its) (ITS).
+
+> **Snapshot-driven tests:** See the [Integration Service](#integration-service) definition above.
+> Creating a [Snapshot](#snapshot) (CI on each build, or `oc create` when testing) is how you
+> ask Konflux to match tenant [ITS](#its) [CRs](#cr) and start the corresponding [PipelineRuns](#pipelinerun).
+
+**Git vs cluster:** YAML under `integration-tests/` is only source; Konflux runs it
```
Comment (ckhordiasma) [inline doc/contributing-konflux-testing-rhoai.md:59]: The gitops is actually defined here - https://gitlab.cee.redhat.com/releng/konflux-release-data/-/tree/main/tenants-config/cluster/stone-prd-rh01/tenants/open-data-hub-tenant

manifests in that folder will get applied to the cluster for ODH. I think it would be good to define a development workflow that is something along these lines:

1. manually `oc apply` temporary pipelineruns, snapshots, and integration test scenarios to the cluster. for the pipelineruns, you can start off by inlining the entire pipeline spec, or reference a pipelinerun in your development fork of odh-konflux-central

2. once you have something working, get it PR'd into odh-konflux-central, and then maybe do some sanity checks with manually `oc apply`ed pipelineruns/snapshots/ITSes that reference odh-konflux-central, main branch, with your changes

3. PR your new ITSes into https://gitlab.cee.redhat.com/releng/konflux-release-data/-/tree/main/tenants-config/cluster/stone-prd-rh01/tenants/open-data-hub-tenant so that they can be managed by argocd

File: doc/contributing-konflux-testing-rhoai.md (hunk 4 above)
Comment (manosnoam) [reply doc/contributing-konflux-testing-rhoai.md]: Replaced the generic konflux-ci.dev link with the correct Konflux cluster-info page.

File: doc/contributing-konflux-testing-rhoai.md (hunk 5 above)
Comment (manosnoam) [reply doc/contributing-konflux-testing-rhoai.md:59]: Fixed the paragraph to explain that ITS CRs are managed by ArgoCD from the konflux-release-data GitLab repo, and added the 3 steps development workflow.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+              # Trust the Red Hat internal CA chain so we don't need GIT_SSL_NO_VERIFY.
+              if [ -d /etc/pki/ca-trust/source/anchors ]; then
+                update-ca-trust 2>/dev/null || true
+              fi
+              # Fallback: if the host still isn't trusted (e.g. missing CA bundle
+              # in the image), allow the clone but warn loudly.
+              if ! git ls-remote --exit-code "${REPO_URL}" HEAD &>/dev/null; then
+                echo "⚠ TLS verification failed for ${REPO_URL} — falling back to GIT_SSL_NO_VERIFY"
+                export GIT_SSL_NO_VERIFY=true
+              fi
```
Comment (aloganat) [inline integration-tests/olminstall/olminstall-smoke-pipeline.yaml]: **TLS verification fallback to `GIT_SSL_NO_VERIFY` on the olminstall clone path**

The `clone-olminstall` step tries `update-ca-trust` but falls back to `GIT_SSL_NO_VERIFY=true` when the internal GitLab host isn't trusted. The scripts cloned from this repo (`install-and-verify.sh`, utilities in `utils/`) are then `exec`'d with the ephemeral cluster's `KUBECONFIG` and mounted Quay pull-secret in scope.

A MITM on this clone path could replace those scripts and pivot to cluster takeover or credential exfiltration (CWE-295/CWE-494). I understand this is pragmatic for internal GitLab, but we should fail closed here rather than silently disable verification.

Suggestion:
- Bake the internal CA bundle into `quay.io/rhoai/rhoai-task-toolset:its` (or mount it as a ConfigMap), so `update-ca-trust` always succeeds.
- If TLS still fails after that, `exit 1` instead of falling back to `GIT_SSL_NO_VERIFY`.

```bash
if ! git ls-remote --exit-code "${REPO_URL}" HEAD &>/dev/null; then
  echo "❌ TLS verification failed for ${REPO_URL}. Mount the internal CA bundle and retry."
  exit 1
fi
```

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+                  value: https://github.com/konflux-ci/build-definitions.git
+                - name: revision
+                  value: main
+                - name: pathInRepo
+                  value: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml
+            params:
+              - name: eaasSpaceSecretRef
+                value: $(tasks.provision-eaas-space.results.secretRef)
+              - name: clusterName
+                value: "$(tasks.provision-cluster.results.clusterName)"
```
Comment (aloganat) [inline integration-tests/olminstall/olminstall-smoke-pipeline.yaml:480]: ** `set -eux` in diagnostics step risks leaking sensitive values to pipeline logs**

The `-x` flag traces every expanded command including `KUBECONFIG` paths, and if any future edit introduces a token-bearing env var (Slack URL, registry credential), it will be echoed verbatim into PipelineRun logs visible in the Konflux UI.

Suggestion: Change to `set -eu` and use explicit `echo` statements for the diagnostic banners. The diagnostic output is already well-structured with clear section headers, so `-x` doesn't add much value here.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml [hunk 8]
```diff
@@ -0,0 +1,582 @@
...
+          operator: in
+          values: ["Failed"]
+      params:
+        - name: OPERATOR_NAMESPACE
+          value: $(params.OPERATOR_NAMESPACE)
+      taskSpec:
+        params:
+          - name: OPERATOR_NAMESPACE
+            type: string
+        volumes:
```
Comment (aloganat) [inline integration-tests/olminstall/olminstall-smoke-pipeline.yaml:462]: ** `collect-diagnostics` finally task may be skipped when it's needed most**

This finally task references `$(tasks.provision-eaas-space.results.secretRef)` and `$(tasks.provision-cluster.results.clusterName)` in its step params. In Tekton v1, if either upstream task was skipped or failed without producing those results, this finally task will be silently skipped — exactly when diagnostics are most needed.

Consider splitting into two finally tasks:
1. A lightweight failure reporter that does **not** reference any task results (always runs on failure)
2. A cluster-specific diagnostics task gated on `provision-cluster` having succeeded (so the result refs are safe)

This way you'll always get at least a notification on failure, even when cluster provisioning itself fails.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml [hunk 9]
```diff
@@ -0,0 +1,582 @@
...
+        # not skip this finally task when install-operator was itself skipped or
+        # failed. "None" is the Tekton status value when the task never ran.
+        - name: TEST_STATUS
+          value: "$(tasks.install-operator.status)"
+        # OPERATOR_VERSION is a best-effort field: pass a safe literal so the
+        # task is never skipped due to a missing result reference.
+        - name: OPERATOR_VERSION
+          value: "(see pipeline run logs)"
+        - name: PIPELINE_RUN_NAME
+          value: $(context.pipelineRun.name)
```
Comment (aloganat) [inline integration-tests/olminstall/olminstall-smoke-pipeline.yaml:600]: **`post-results` task params safely avoid result dereferences — nice work**

I can see the careful NOTE comments explaining why you pass `$(tasks.install-operator.status)` and a literal fallback for `OPERATOR_VERSION` instead of result references. This is the correct pattern for Tekton v1 finally tasks. Well done.

One minor nit: the `FBCF_IMAGE` param here embeds `$(params.SNAPSHOT)` which could be a very large JSON blob. Consider truncating or just passing `$(params.FBCF_COMPONENT_NAME)` alone to keep the Slack message readable.

File: integration-tests/olminstall/scripts/install-and-verify.sh [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+EOF
+
+echo "Waiting for OLM to create the ${OLMINSTALL_CATALOG_NAME} ServiceAccount (up to 2m)..."
+SA_DEADLINE=$(($(date +%s) + 120))
+while [ "$(date +%s)" -lt "$SA_DEADLINE" ]; do
+  oc get sa "${OLMINSTALL_CATALOG_NAME}" -n openshift-marketplace &>/dev/null && break
+  sleep 5
+done
+if ! oc secrets link "${OLMINSTALL_CATALOG_NAME}" "${QUAY_PULL_SECRET_NAME}" -n openshift-marketplace --for=pull 2>/dev/null; then
+  echo "⚠ Could not link ${QUAY_PULL_SECRET_NAME} to ${OLMINSTALL_CATALOG_NAME} SA (SA may not exist yet — non-fatal)"
```
Comment (aloganat) [inline integration-tests/olminstall/scripts/install-and-verify.sh:96]: **`sleep 30` after CatalogSource pod delete is racy**

On a cold EaaS cluster pulling `quay.io/rhoai/...` for the first time, 30 seconds may not be enough for OLM to schedule and start the replacement pod. Since the 15-minute READY wait loop immediately follows and already handles the "pod not yet running" case, this fixed sleep adds latency without adding reliability.

Suggestion: Either remove the sleep entirely (the READY poll handles it), or replace with a deterministic wait:
```bash
oc wait --for=delete pod -n openshift-marketplace \
  -l "olm.catalogSource=${OLMINSTALL_CATALOG_NAME}" --timeout=60s || true
```

File: integration-tests/olminstall/scripts/install-and-verify.sh [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+echo " ODH/RHOAI Operator Installation"
+echo " FBCF:      ${FBCF_IMAGE}"
+echo " Channel:   ${UPDATE_CHANNEL}"
+echo " Operator:  ${OPERATOR_NAME} -> ${OPERATOR_NAMESPACE}"
+echo "========================================="
+
+oc version || { echo "❌ Cannot connect to cluster"; fail; }
+
+oc create namespace "${OPERATOR_NAMESPACE}" --dry-run=client -o yaml | oc apply -f -
+
```
Comment (aloganat) [inline integration-tests/olminstall/scripts/install-and-verify.sh:63]: ** FBCF_IMAGE validation is good — consider also validating it looks like a valid image ref**

Good that you validate `FBCF_IMAGE` against `^[A-Za-z0-9./_:@-]+$` to prevent YAML injection. The regex is correct for standard OCI image refs.

However, you might also want to enforce it contains at least one `/` and an `@sha256:` or `:tag` to catch obvious misconfigurations early (e.g., someone passing just a tag name).

File: integration-tests/olminstall/scripts/install-and-verify.sh [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+}
+
+# Match the CSV olminstall actually installed (OLMINSTALL_OPERATOR may be rhods-operator
+# when OPERATOR_NAME is opendatahub-operator and we use the rhods-operator manifest fallback).
+CSV_VERSION=$(oc get csv -n "${OPERATOR_NAMESPACE}" -o json 2>/dev/null \
+  | jq -r --arg op "${OLMINSTALL_OPERATOR}" '
+      [ .items[]
+        | select(.status.phase == "Succeeded")
+        | select((.metadata.name // "" | startswith($op))
+                 or (.spec.displayName // "" | test($op; "i"))) ]
```
Comment (aloganat) [inline integration-tests/olminstall/scripts/install-and-verify.sh:250]: ** CSV version extraction could match the wrong operator**

The jq filter selects CSVs where `metadata.name` starts with `$op` OR `spec.displayName` case-insensitively matches `$op`. In a namespace with multiple operators, this could match an unrelated CSV that happens to have a superset name (e.g., `rhods-operator-helper` matching `rhods-operator`).

Suggestion: Anchor the name match more precisely:
```jq
select(.metadata.name | test("^" + $op + "\\."))
```
This ensures you match `rhods-operator.X.Y.Z` but not `rhods-operator-something-else`.

File: integration-tests/olminstall/scripts/install-and-verify.sh [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+OLMINSTALL_MANIFEST="${OLMINSTALL_DIR}/resources/install-${OPERATOR_NAME}.yaml"
+if [[ ! -f "${OLMINSTALL_MANIFEST}" ]]; then
+  FALLBACK_MANIFEST="${OLMINSTALL_DIR}/resources/install-rhods-operator.yaml"
+  if [[ -f "${FALLBACK_MANIFEST}" ]]; then
+    echo "⚠ Manifest install-${OPERATOR_NAME}.yaml not found — using rhods-operator manifest"
+    OLMINSTALL_MANIFEST="${FALLBACK_MANIFEST}"
+    OLMINSTALL_OPERATOR="rhods-operator"
+  else
+    fail "❌ No olminstall manifest found for operator ${OPERATOR_NAME}"
+  fi
```
Comment (aloganat) [inline integration-tests/olminstall/scripts/install-and-verify.sh:227]: **The `sed` replacement on the olminstall manifest namespace is too broad**

This `sed` command replaces `redhat-ods-operator` only when it appears as a YAML `namespace:` value (anchored with `^\s*namespace:\s*`), which is a good improvement over a global replace. However, if the manifest has multiple resources with `namespace: redhat-ods-operator` in different contexts (e.g., a RoleBinding targeting a different namespace), this could still cause issues.

Consider using `yq` for a structural edit:
```bash
yq e '(select(.kind == "Namespace" or .kind == "OperatorGroup" or .kind == "Subscription") | .metadata.namespace) = env(OPERATOR_NAMESPACE)' -i "${OLMINSTALL_MANIFEST}"
```

Not blocking, since the current regex is reasonably safe for the known manifest structure.

File: integration-tests/olminstall/run-olminstall.sh [hunk 14]
```diff
@@ -0,0 +1,493 @@
...
+  | jq -r --arg app "${APP}" '
+      .items[]
+      | select(.metadata.name | test("olminstall"))
+      | select(.status.completionTime == null)
+      | select(
+          (.metadata.labels["appstudio.openshift.io/application"] // "") == $app
+          or (.spec.params // [] | map(select(.name == "SNAPSHOT")) | length == 0)
+        )
+      | .metadata.name' \
+  | head -1 || true)
```
Comment (aloganat) [inline integration-tests/olminstall/run-olminstall.sh:183]: ** Auto-attach to running PipelineRun could cross-attach to a different user's run**

The jq filter selects any incomplete PipelineRun with "olminstall" in its name matching the same `APP` label. If two users run `run-olminstall.sh` against the same namespace/app around the same time, one might attach to the other's run instead of creating a new one.

The README already documents this race condition, which is good. But consider adding a check on the Snapshot name or creator annotation to verify the run was triggered by the current user before attaching. This would make concurrent use safer.

Alternatively, you could add a `--no-reuse` flag to force creating a new run even if one exists.

File: integration-tests/olminstall/run-olminstall.sh [hunk 15]
```diff
@@ -0,0 +1,493 @@
...
+  done
+  if [ "$(date +%s)" -ge "${POLL_DEADLINE}" ]; then
+    echo "❌ Polling timed out before pipeline reached a terminal state (${MAX_WAIT_SECONDS}s)"
+    PIPELINE_EXIT=1
+  fi
+fi
+
+# ── Summary ───────────────────────────────────────────────────────────────────
+IFS=$'\t' read -r FINAL_CSTAT FINAL_STATUS <<< "$(pr_succeeded_state "${PR}" "${NAMESPACE}")"
+OPERATOR_VERSION=$(sed -n 's/.*Operator version[[:space:]]*:[[:space:]]*\([^[:space:]]*\).*/\1/p' "${LOG_FILE}" 2>/dev/null | tail -1 || true)
```
Comment (aloganat) [inline integration-tests/olminstall/run-olminstall.sh:482]: **`grep -oP` is GNU-only and fails silently on macOS**

This script is documented as a developer laptop entrypoint, but `grep -oP` (PCRE) is not available on macOS's BSD grep. The extraction silently returns empty, so the summary just omits the version with no error.

Use a portable alternative:
```bash
OPERATOR_VERSION=$(sed -n 's/.*Operator version[[:space:]]*:[[:space:]]*\([^[:space:]]*\).*/\1/p' "${LOG_FILE}" 2>/dev/null | tail -1 || true)
```

File: integration-tests/olminstall/run-olminstall.sh [hunk 16]
```diff
@@ -0,0 +1,493 @@
...
+    if [[ -n "${ODH_PARAM_OVERRIDES}" ]]; then
+      yq e '.spec.params += [{"name":"OPERATOR_NAME","value":"opendatahub-operator"}]' -i "${ITS_APPLY_TMP}"
+      yq e '.spec.params += [{"name":"OPERATOR_NAMESPACE","value":"opendatahub-operators"}]' -i "${ITS_APPLY_TMP}"
+      yq e '.spec.params += [{"name":"FBCF_COMPONENT_NAME","value":"odh-operator-catalog"}]' -i "${ITS_APPLY_TMP}"
+    fi
+    echo "  ITS overrides: resolverRef=${KONFLUX_REPO_OVERRIDE:-<default>}@${KONFLUX_BRANCH_OVERRIDE:-<default>}" \
+         " SCRIPTS_REPO=${KONFLUX_REPO_OVERRIDE:-<default>}@${KONFLUX_BRANCH_OVERRIDE:-<default>}" \
+         " UPDATE_CHANNEL=${UPDATE_CHANNEL_OVERRIDE:-<pipeline default>}" \
+         " PRODUCT=${PRODUCT}"
+    oc apply -n "${NAMESPACE}" -f "${ITS_APPLY_TMP}" 2> >(grep -v "^Warning" >&2)
```
Comment (aloganat) [inline integration-tests/olminstall/run-olminstall.sh]: `oc apply` stderr filtering hides real failures

The `2> >(grep -v "^Warning" >&2)` pattern strips warnings from stderr, but `|| true` is **not** used here (which is actually good, you do check the exit code). However, process substitution (`>(...)`) can race with the parent process — `oc apply` might exit before the `grep` subprocess finishes reading stderr, potentially losing error messages.

A more robust pattern:
```bash
oc apply -n "${NAMESPACE}" -f "${ITS_APPLY_TMP}" 2>&1 | grep -v "^Warning" >&2
APPLY_STATUS=${PIPESTATUS[0]}
[[ ${APPLY_STATUS} -ne 0 ]] && { echo "❌ ITS apply failed"; exit 1; }
```

File: integration-tests/olminstall/scripts/patch-cluster-pull-secret.sh [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+  echo "❌ No quay.io/rhoai auth token found in ${QUAY_SECRET}"
+  exit 1
+fi
+
+QUAY=$(echo "$QUAY" | jq --arg a "$QUAY_AUTH" '.auths["quay.io"] = {"auth": $a}')
+
+echo "Patching cluster global pull secret with quay.io/rhoai credentials..."
+EXISTING=$(oc get secret/pull-secret -n openshift-config \
+  -o jsonpath='{.data.\.dockerconfigjson}' | base64 -d)
+MERGED=$(jq -s '.[0] * {auths: ((.[0].auths // {}) * (.[1].auths // {}))}' \
```
Comment (aloganat) [inline integration-tests/olminstall/scripts/patch-cluster-pull-secret.sh:51]: Ignore this comment.

File: integration-tests/olminstall/test-pipelinerun.yaml [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+    - name: FBCF_COMPONENT_NAME
+      value: odh-operator-catalog
+    - name: UPDATE_CHANNEL
+      value: odh-stable
+    - name: OPERATOR_NAMESPACE
+      value: opendatahub-operators
+    - name: OPERATOR_NAME
+      value: opendatahub-operator
+    - name: OLMINSTALL_REPO_URL
+      value: https://gitlab.cee.redhat.com/data-hub/olminstall.git
```
Comment (aloganat) [inline integration-tests/olminstall/test-pipelinerun.yaml:43]: ** Missing `QUAY_PULL_SECRET_NAME` param for ODH test runs**

This test PipelineRun sets ODH-specific params (`OPERATOR_NAME=opendatahub-operator`, etc.) but doesn't set `QUAY_PULL_SECRET_NAME`. The pipeline default is `rhoai-quay-secret`, which won't exist in an ODH-only namespace.

Considering adding:
```yaml
    - name: QUAY_PULL_SECRET_NAME
      value: odh-quay-secret
```

Or document that the user needs to create/rename the secret to match the pipeline default.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml [hunk 19]
```diff
@@ -0,0 +1,582 @@
...
+          - name: FBCF_IMAGE
+            description: Extracted FBCF image URL
+        steps:
+          - name: parse-snapshot
+            image: quay.io/konflux-ci/konflux-test:stable
+            env:
+              - name: SNAPSHOT
+                value: $(params.SNAPSHOT)
+              - name: COMPONENT_NAME
+                value: $(params.COMPONENT_NAME)
```
Comment (aloganat) [inline integration-tests/olminstall/olminstall-smoke-pipeline.yaml:109]: ** Image tag `quay.io/rhoai/rhoai-task-toolset:its` is pinned by digest — good**

I see this image is pinned with sha256 across all steps. This is good practice for supply-chain integrity and reproducibility. The `quay.io/konflux-ci/konflux-test:stable` used in `extract-fbcf-image` and `send-notification` steps is still a floating tag though. Consider pinning those to a digest as well for consistency.

**Review by aloganat — COMMENTED** [review]
Thanks for the thorough work here, Noam. This is a substantial addition .

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml (hunk 6 above)
Comment (manosnoam) [reply integration-tests/olminstall/olminstall-smoke-pipeline.yaml]: Fixed: replaced global `GIT_SSL_NO_VERIFY=true` with a host-scoped `git config http.<host>.sslVerify=false` for `gitlab.cee.redhat.com` only. All other hosts now fail with `exit 1`. TODO left in code to remove this once the RH IT CA is in the task image.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml (hunk 7 above)
Comment (manosnoam) [reply integration-tests/olminstall/olminstall-smoke-pipeline.yaml:480]: Fixed: changed to `set -eu` in the diagnostics step.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml (hunk 8 above)
Comment (manosnoam) [reply integration-tests/olminstall/olminstall-smoke-pipeline.yaml:462]: Fixed: added `$(tasks.provision-cluster.status) == Succeeded` gate so Tekton won't skip the task due to missing cluster result refs. `post-results` still fires on all failures via task status, not result refs.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml (hunk 9 above)
Comment (manosnoam) [reply integration-tests/olminstall/olminstall-smoke-pipeline.yaml:600]: Acknowledged.

File: integration-tests/olminstall/scripts/install-and-verify.sh (hunk 10 above)
Comment (manosnoam) [reply integration-tests/olminstall/scripts/install-and-verify.sh:96]: Fixed: replaced `sleep 30` with `oc wait --for=delete pod -l olm.catalogSource=... --timeout=60s`.

File: integration-tests/olminstall/scripts/install-and-verify.sh (hunk 11 above)
Comment (manosnoam) [reply integration-tests/olminstall/scripts/install-and-verify.sh:63]: Acknowledged. Can add stricter image-ref validation as a follow-up.

File: integration-tests/olminstall/scripts/install-and-verify.sh (hunk 12 above)
Comment (manosnoam) [reply integration-tests/olminstall/scripts/install-and-verify.sh:250]: Fixed: CSV jq filter now uses `OLMINSTALL_OPERATOR` (the installed package name) instead of `OPERATOR_NAME`.

File: integration-tests/olminstall/scripts/install-and-verify.sh (hunk 13 above)
Comment (manosnoam) [reply integration-tests/olminstall/scripts/install-and-verify.sh:227]: Fixed: `sed` pattern is now anchored to the `namespace:` YAML field line only.

File: integration-tests/olminstall/run-olminstall.sh (hunk 14 above)
Comment (manosnoam) [reply integration-tests/olminstall/run-olminstall.sh:183]: Fixed: reuse logic now filters by `appstudio.openshift.io/application` label matching `APP`.

File: integration-tests/olminstall/run-olminstall.sh (hunk 15 above)
Comment (manosnoam) [reply integration-tests/olminstall/run-olminstall.sh:482]: Fixed: replaced `grep -oP` with `sed -n 's/.../\1/p'`.

File: integration-tests/olminstall/run-olminstall.sh (hunk 16 above)
Comment (manosnoam) [reply integration-tests/olminstall/run-olminstall.sh]: Fixed: switched to `2>&1 | grep -v '^Warning' >&2` and check `PIPESTATUS[0]`.

File: integration-tests/olminstall/scripts/patch-cluster-pull-secret.sh (hunk 17 above)
Comment (manosnoam) [reply integration-tests/olminstall/scripts/patch-cluster-pull-secret.sh:51]: ack (`FBCF_IMAGE` param now passes just `$(params.FBCF_COMPONENT_NAME)`)

File: integration-tests/olminstall/test-pipelinerun.yaml (hunk 18 above)
Comment (manosnoam) [reply integration-tests/olminstall/test-pipelinerun.yaml:43]: Fixed: added `QUAY_PULL_SECRET_NAME: odh-quay-secret` to `test-pipelinerun.yaml`.

File: integration-tests/olminstall/olminstall-smoke-pipeline.yaml (hunk 19 above)
Comment (manosnoam) [reply integration-tests/olminstall/olminstall-smoke-pipeline.yaml:109]: Fixed: pinned `quay.io/konflux-ci/konflux-test:stable` to digest `sha256:09328e5f47da168ffe951ec9c4242cd6761e61e0a235df1bba3f5158e757446d`.

**Review by aloganat — APPROVED** [review]
